File: Miscellaneous/Golang/344. Reverse String.go
package main

import "fmt"

func main() {

	str := []string{"H", "a", "n", "n", "a", "h"}
	result := reverseString(str)
	fmt.Println(result)
}

func reverseString(str []string) []string {
	for i, j := 0, len(str)-1; i < j; i, j = i+1, j-1 {
		str[i], str[j] = str[j], str[i]
	}
	return str
}

File: Miscellaneous/Python/344. Reverse String.py
class Solution:
    def reverseString(self, str):
        i, j = 0, len(str)-1

        while i < j:
            str[i], str[j] = str[j], str[i]
            i += 1
            j -= 1
        
        return str

obj = Solution()
str = ["H","a","n","n","a","h"]
ans = obj.reverseString(str)
print(ans)